plugins/woocommerce/includes/class-wc-countries.php

```diff
@@ -287,34 +287,33 @@ public function get_base_postcode() {
 	 * @return array
 	 */
 	public function get_allowed_countries() {
-		if ( 'all' === get_option( 'woocommerce_allowed_countries' ) ) {
-			return apply_filters( 'woocommerce_countries_allowed_countries', $this->countries );
-		}
+		$countries = $this->countries;
 
 		if ( 'all_except' === get_option( 'woocommerce_allowed_countries' ) ) {
 			$except_countries = get_option( 'woocommerce_all_except_countries', array() );
 
-			if ( ! $except_countries ) {
-				return $this->countries;
-			} else {
-				$all_except_countries = $this->countries;
+			if ( $except_countries ) {
 				foreach ( $except_countries as $country ) {
-					unset( $all_except_countries[ $country ] );
+					unset( $countries[ $country ] );
 				}
-				return apply_filters( 'woocommerce_countries_allowed_countries', $all_except_countries );
 			}
-		}
-
-		$countries = array();
+		} elseif ( 'specific' === get_option( 'woocommerce_allowed_countries' ) ) {
```

Comment:
Would it be better to get this option once and assign it to a variable (same for `woocommerce_ship_to_countries`)? It would be a small performance benefit, I know it wouldn't be huge, but it would save a few cycles.

Is the readability/maintainability penalty worth it?